src/algorithms/reco/FarForwardNeutronReconstruction.cc

```diff
@@ -7,20 +7,27 @@
 #include <edm4hep/Vector3f.h>
 #include <math.h>
 #include <gsl/pointers>
+#include <vector>
 
 /**
- Creates a "neutron candidate" Reconstructed Particle consisting of all clusters in a given ClusterCollection.
- Its energy is the sum of the energies of the constituent clusters, and its direction is the direction from the
-origin to the position of the most energetic cluster
+ Creates a "neutron candidate" Reconstructed Particle consisting of all clusters in a
+ given ClusterCollection.  Its energy is the sum of the energies of the constituent clusters
+ times a correction factor, and its direction is the direction from the origin to the position
+ of the most energetic cluster.  The correction factor is given by 1/(1+c[0]+c[1]/sqrt(E)+c[2]/E),
+ where c is the coefficients and E is the uncorrected energy in GeV.  This form was chosen
+ empirically based on the discrepancies in single-neutron MC simulations between the uncorrected
+ reconstructed energies and the truth energies of the neutrons.  
  */
 
 namespace eicrecon {
 
     void FarForwardNeutronReconstruction::init() {  }
```

Comment:
Would you mind adding a check that coeffs has at least 3 elements?

Reply:
I agree with this; should this be in the init function or when processing events?   Should it shut down the process or just give a warning?  Do you have an example of what other algorithms do when checking for valid/invalid configuration values?
Thanks,
Sebouh

Reply:
ok, found that other algorithms throw a JException with a description of what went wrong with improper configuration.  I have edited the algorithm accordingly